### DISCUSSION TITLE
Help appreciated for compatibility Gedcom 7

### DISCUSSION THREAD

**Discussion by mariannevanharten** [FamilySearch/GEDCOM#217] [Q&A]
I am trying to make my gedcom file compatible to Gedcom 7.
Some issues are very hard for me to solve. I have read the website https://gedcom.io/specifications/FamilySearchGEDCOMv7.html
I hope here is a person who can help me. I am using Chronoplex' Gedcom Validator to check my gedcom.

Invalid tag structure. Tag 'RESN' is not   permitted under tag 'CAUS'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'CHIL'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'CHIL'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'DATE'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'DATE'.
Invalid tag structure. Tag 'CTRY' is not   permitted under tag 'DEAT'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'DEAT'.
Invalid tag structure. Tag 'CTRY' is not   permitted under tag 'EMIG'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'EMIG'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'FAMC'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'FAMS'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'GIVN'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'HUSB'.
Invalid tag structure. Tag 'DATE' is not   permitted under tag 'NAME'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'NAME'.
Invalid tag structure. Tag 'REPO' is not   permitted under tag 'NOTE'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'NOTE'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'OBJE'.
Invalid tag structure. Tag 'CTRY' is not   permitted under tag 'OCCU'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'OCCU'.
Invalid tag structure. Tag 'DATE' is not   permitted under tag 'PHON'.
Invalid tag structure. Tag 'NOTE' is not   permitted under tag 'PHON'.
Invalid tag structure. Tag 'CTRY' is not   permitted under tag 'RESI'.
Invalid tag structure. Tag 'DATE' is not   permitted under tag 'RESN'.
Invalid tag structure. Tag 'NOTE' is not   permitted under tag 'RESN'.
Invalid tag structure. Tag 'DATE' is not   permitted under tag 'SEX'.
Invalid tag structure. Tag 'TYPE' is not   permitted under tag 'SEX'.
Invalid tag structure. Tag 'ASSO' is not   permitted under tag 'SOUR'.
Invalid tag structure. Tag 'NAME' is not   permitted under tag 'SOUR'.
Invalid tag structure. Tag 'PLAC' is not   permitted under tag 'SOUR'.
Invalid tag structure. Tag 'REPO' is not   permitted under tag 'SOUR'.
Invalid tag structure. Tag 'QUAY' is not   permitted under tag 'SURN'.
Invalid tag structure. Tag 'NOTE' is not   permitted under tag 'TYPE'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'TYPE'.
Invalid tag structure. Tag 'RESN' is not   permitted under tag 'WIFE'.
'NONE' is not a valid <List:Enum>   RESN.

Many thanks in advance for your help.

Best regards,
Marianne

**Comment by clarkegj** [comment]
Chronoplex has not released a GEDCOM 7 Validator so the invalid tags mags sense.

**Reply by ungeahnt** [reply]
@clarkegj? 
see https://chronoplexsoftware.com/gedcomvalidator/index.htm

> Validation
> Check the [validity](https://chronoplexsoftware.com/gedcomvalidator/help/gedcomvalidator.htm) of GEDCOM 5.5, 5.5.1, 5.6, and 7.x files.  

How well it works, I can not say, because I do not work with it.

**Comment by Norwegian-Sardines** [comment]
Marianne,

I took a quick look at a few of the errors generated:

SEX does not allow a subtag of DATE.
A NOTE tag can not have a REPO or RESN subtag.

so I’m guessing that the validator is telling you some truths about you GEDCOM.

**Comment by tychonievich** [comment]
- You appear to be missing an intermediate `ADDR` for `CTRY`: instead of `EMIG`.`CTRY` do `EMIG`.`ADDR`.`CTRY`
- `RESN NONE` is not supported, but presumably means the same thing as the lack of a `RESN` tag (i.e., no restrictions recorded)
- `SOUR`.`REPO` is supported for source records, but not source references. Try moving your `XYZ`.`SOUR`.`REPO` to the source record the `SOUR` points to
- I'm guessing `SOUR`.`NAME` was supposed to be `SOUR`.`TITL`?
- The others I don't immediately see a clean solution to: they appear to be using standard tags in non-standard locations. If you really wanted those tags to be there, you can replace them with extension tags (e.g. change `DATE`.`RESN` to `DATE`.`_RESN`) but while that will validate, it's not likely to be understood by other applications. If you want to provide more details about why one or another sturcture was wanted, we might be able to point you toward a supported way to represent that data.

**Comment by fisharebest** [comment]
> `RESN NONE` is not supported, but presumably means the same thing as the lack of a `RESN` tag (i.e., no restrictions recorded)

Not quite...  As the probable author of that tag(!), see #220
